crates/ruff_workspace/src/settings.rs

```diff
@@ -124,6 +126,8 @@ pub struct FormatterSettings {
     pub magic_trailing_comma: MagicTrailingComma,
 
     pub line_ending: LineEnding,
+
+    pub docstring_code: DocstringCode,
```

Comment:
I feel like we're going to end up bikeshedding on this name... a few ideas for consideration:

- `include_docstrings=true` (I like this but it doesn't convey that we're describing _code_ in docstrings)
- `docstrings=true`
- `include_embedded_code=true`
- `include_docstring_code=true`
- `include_docstring_snippets=true`

Reply:
Yeah that was why I just picked the first thing that came to mind.

Of your list, I like `include_docstring_code` the best. But I think I do actually like `docstring_code` the best. It kind of feels like the word "include" is a little redundant? Looking at the names of other options, it doesn't look like there is any strong precedent here. Although using "include" here could start a precedent. (I don't know whether we plan on adding a ton of options or not though.)

Reply:
I was okay with this on the first read, but on a second, I think `docstring_code = true | false` doesn't have enough meaning alone. Perhaps `format_docstring_code` would make sense? I also don't mind `include_docstring_code` but it's not necessarily clear what inclusion means.

Reply:
Yeah I hadn't gone with `format_docstring_code` since the option exists in the context of a formatter. But I think I do like that better than `include_docstring_code`.

Reply:
I still feel like none of these are quite right (the docstring itself _is_ code)... It's a really hard thing to convey succinctly. I wish we could do `format_doctests` but this needs to generalize to embedded Markdown and rST. Is `format_docstring_snippets` clearer or worse? (At the end of the day, I can live with `format_docstring_code`.)

Reply:
PR for the user facing change created (and removed from this PR): https://github.com/astral-sh/ruff/pull/8854